### PR TITLE
Plugin check update #7186

### PR DESCRIPTION
**Pull request by ivandrofly**
Add the capability to also check for plugin update when checking for SE updates via 
`Help/Check for updates`

### CODE REVIEW

File: src/Test/Core/PluginUpdateCheckerTest.cs [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+namespace Test.Core
+{
+    [TestClass]
+    public class PluginUpdateCheckerTest
+    {
+        [TestMethod]
+        public async Task CheckAsyncTest()
+        {
+            var pluginPath = Environment.GetEnvironmentVariable("plugin_directory", EnvironmentVariableTarget.User);
```
Comment (ivandrofly) [inline src/Test/Core/PluginUpdateCheckerTest.cs:14]: Looking for a solution to this...
- Built one of the plugin fiel with decreased version!

Should we let the test cases call internet?

**Comment by ivandrofly** [comment]
UI demo

image

**Comment by ivandrofly** [comment]
This class will also be reused in PluginGet window

**Comment by niksedk** [comment]
Something like this?
image

>This class will also be reused in PluginGet window
Surely :)

Nice with nsubstitute and not moq - what a mess...

Perhaps a bit simpler structure / less code. Too much inheritance for my taste...

**Comment by niksedk** [comment]
I can also merge this a refactor a bit?

**Comment by ivandrofly** [comment]
> I can also merge this a refactor a bit?

Sure, go for it :)

**Comment by ivandrofly** [comment]
> Nice with nsubstitute and not moq - what a mess...

Not a fan :D

**Comment by niksedk** [comment]
Main branch + beta updated: https://github.com/SubtitleEdit/subtitleedit/releases/download/3.6.13/SubtitleEditBeta.zip

Could you give it a test run?

**Comment by ivandrofly** [comment]
> Main branch + beta updated: https://github.com/SubtitleEdit/subtitleedit/releases/download/3.6.13/SubtitleEditBeta.zip
> 
> Could you give it a test run?

Sure :)

**Pull request merged**